### ISSUE TITLE
Hotgmck cheat menu

### ISSUE THREAD

**Issue by mahoneyt944** [libretro/mame2003-plus-libretro#1489]
Doesn't display correctly

@MistyDreams opened an issue for this.

It looks like the string is read in properly just not being formated correctly in the interface.
![Screenshot_20221202-082733](https://user-images.githubusercontent.com/49591133/205303846-1525fb3f-cd5b-4088-b849-80d67fcbcb8e.png)
![Screenshot_20221202-082717](https://user-images.githubusercontent.com/49591133/205303847-dbdb22fc-cc2e-46a8-8b0f-ecb699a1563a.png)

**Comment by mahoneyt944** [comment]
Maybe an issue with a newline or return? Seems to jam it all in one line.

**Comment by mahoneyt944** [comment]
The rzip file has to be manually deleted, if trying to edit the dat. A new one will be automatically generated. When not found.

The first comment is this below, isn't the limit 255? If not what is the limit?
```
[ The winning hand cheat must be activated when there are 14 tiles in your hand, then press the RON   ]
```

**Comment by mahoneyt944** [comment]
The full comment is read in though, just not showing correctly on that page?

**Comment by mahoneyt944** [comment]
And the issue with crc checks with rzip is youd be checking the initial files crc and once edited, the file won't have the same crc anymore, and therefore would rebuild everytime you load a game, you'd have to save/ update this crc to reference on the next load but that seems like a hassle.

**Comment by mahoneyt944** [comment]
Ah yes the dual screen mode is probably why, it's formating the width wrong.

**Comment by arcadez2003** [comment]
Sorry to but in here but it's related to these games, so long story short im about to commit an input change in this
driver which will make Lode Runner The Dig Fight boot to the world English Region by default which is handy for game
modes and types.

This could be done before by switching the region manually but it was a pain to traverse the service mode and this had
to be done each time you booted the game as the settings never stuck, so im adding a new eeprom setting and a new
region dip switch to the inputs.

@mahoneyt944 can you cast your eyes over the commit im going to push to see if you think i've done it correctly
certainly the game is booting to the English region and all the other inputs seem fine it's just there are so many fake
inputs and i had to alter the original code from the tagging method...

So here is the change.

In the eeprom
return readinputportbytag("JP4")<<16;

Inputs for taisen and hotgimmick extra input and fake port
PORT_START_TAG("JP4")/* jumper pads 'JP4' on the PCB */
	UNUSED_PORT

inputs for Lode Runner new port for the region
PORT_START_TAG("JP4")/* jumper pads 'JP4' on the PCB */
//  1-ON,2-ON,3-ON,4-ON  --> Japanese
//  1-ON,2-ON,3-ON,4-OFF --> English
	PORT_DIPNAME( 0x03, 0x01, DEF_STR( Region ) )
	PORT_DIPSETTING(    0x00, "Japan (Shows Version Number)" )
	PORT_DIPSETTING(    0x01, "World (Does Not Show Version Number)" )

Basically i've counted the input ports for Lode Runner including the faked ones and came up with 9 so the extra one is 10
using the non tagged method i think hence.......

	return readinputport(10)<<16;
	PORT_START/* jumper pads 'JP4' on the PCB */
/*  1-ON,2-ON,3-ON,4-ON  --> Japanese
    1-ON,2-ON,3-ON,4-OFF --> English */
	PORT_DIPNAME( 0x03, 0x01, "Region" )
	PORT_DIPSETTING(    0x00, "Japan (Shows Version Number)" )
	PORT_DIPSETTING(    0x01, "World (Does Not Show Version Number)" )
INPUT_PORTS_END

Whatya think correct.??

**Comment by arcadez2003** [comment]
> this is a dual screen game its only showing one screen though just checked in mame078 that shows the two screens. That might increase the columns available.

I turned on the single screen hack some years back so maybe that is affecting things also.

**Comment by mahoneyt944** [comment]
@arcadez2003 that looks ok to me.

**Comment by arcadez2003** [comment]
> @arcadez2003 that looks ok to me.

Thanks man i know your good at the inputs hence i thought it worth asking ya cheers

**Comment by arcadez2003** [comment]
> > I turned on the single screen hack some years back so maybe that is affecting things also.
> 
> I dont think many play mahjong games to be honest its just the best example of the issue. I dont think changing to dual screen would help this issue looking at the code. Its the menu osd code that would need updated or the cheat file needing edited since its probably form a newer mame.

Same here the only game in the driver worth bothering about is the Lode Runner update which is far better in single screen
than two smaller ones IMO.

**Comment by mahoneyt944** [comment]
Does the osd have a dual screen flag or anything? I guessing the width needed divided by 2 for dual screen only.

**Comment by mahoneyt944** [comment]
>https://github.com/libretro/mame2003-plus-libretro/blob/12b4c90b9146d9744b7cf6c98a37100767369e5e/src/mame2003/video.c#L84-L86

That's the width for the ui box itself which seems correct, it's the text within the box that's bigger than it should be. Seems like it could be specific to comments too.

**Comment by mahoneyt944** [comment]
Current mame2003 acts the same way as 2003+. Definitely need to look into what change was made. Here's current mame2003 with mspacman and hotgmck. Note that without loading the cheat file at all, the comment to unzip it and place it in the directory is still expanded too far. Seems the issue is formatting comments. This works properly with inputs, using the "..."  
 
![Screenshot_20221203-122335](https://user-images.githubusercontent.com/49591133/205453643-81a355d0-4c31-4c00-93e1-615f49d3c799.png)
![Screenshot_20221203-122245](https://user-images.githubusercontent.com/49591133/205453644-265bc33d-cfa7-46bb-b523-8a9e9cd609e7.png)
![Screenshot_20221203-122058](https://user-images.githubusercontent.com/49591133/205453645-0655166c-2887-4186-9a41-a664e9365ee3.png)

**Comment by mahoneyt944** [comment]
What add/edit cheats always center aligned?

**Comment by mahoneyt944** [comment]
Do you have a screen capture from mame078 to reference?

**Comment by mahoneyt944** [comment]
Ok, I reverted cheat.c to 078. It's not this file causing the issue. Must be in the menu code specifically

**Comment by mahoneyt944** [comment]
I pulled it back to the first commit with this file.and it still has that issue https://github.com/libretro/mame2003-plus-libretro/compare/T2

I think it might be an issue with the usrintf

**Comment by mahoneyt944** [comment]
I don't have an Android port of Xtreme to try.

**Comment by mahoneyt944** [comment]
Not at this time.

**Comment by mahoneyt944** [comment]
Well it's effecting a lot of games, not necessarily game breaking but an annoyance for sure. Just need to investigate on it. Probably something dumb.

**Comment by mahoneyt944** [comment]
@KMFDManic could you load mspacman in mame2003 Xtreme and navigate to the cheat menu, under add/edit cheats and screenshot it?
Looks like this in 2003+
![Screenshot_20221203-152157](https://user-images.githubusercontent.com/49591133/205460485-937d7b2c-7347-4a1c-ae50-a62f2981bc41.png)

**Comment by mahoneyt944** [comment]
Hmm, well there should be a simple check on the comment length in relation to the UI's `x` size / width.... Since it has to fit within those parameters. Maybe this was missed somewhere?

**Comment by mahoneyt944** [comment]
Which scan? Sscanf?

**Comment by mahoneyt944** [comment]
Doesn't 078 use those as well without this issue?

**Comment by mahoneyt944** [comment]
It's worth a test, can we test an alternate approach to roll that out?

**Comment by mahoneyt944** [comment]
Though google shows
![Screenshot_20221203-225459](https://user-images.githubusercontent.com/49591133/205473551-698a0ee2-94da-4770-bb54-516ad9ae5094.png)

**Comment by mahoneyt944** [comment]
That's fair, but I don't think that's the issue here. We aren't getting bad data, it's just not formated correctly in the ui. All the data is there.
![hotgmck-221203-230701](https://user-images.githubusercontent.com/49591133/205473909-0987b317-392f-4382-b480-8b5370ea48d3.png)

**Comment by grant2258** [comment]
> Hmm, well there should be a simple check on the comment length in relation to the UI's `x` size / width.... Since it has to fit within those parameters. Maybe this was missed somewhere?
> 
> I'm guessing the displayed comment text is a bitmap? Which needs a fix on the parameters when it's drawn?

@mahoneyt944 libretro runs at the arcades resolution, Normal mame ran on the native resolution and upscaled we cant really do that all you can do is increase the resolution and add an helpers needed to resize the ui. Just setting a higher res would mess the shaders up. Mame current has the same issue with the ui size when set to arcade resolution but it also lets you pick higher resolutions.
